### PR TITLE
Fix warnings in both compiler and stdlib

### PR DESCRIPTION
**Pull request by PavelPunegov**

### CODE REVIEW

File: runtime/src/main/kotlin/kotlin/collections/Arrays.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@ import kotlin.util.sortArray
 public actual inline fun <reified T> Array<out T>?.orEmpty(): Array<out T> = this ?: emptyArray<T>()
 
 
-@PublishedApi internal inline fun checkCopyOfRangeArguments(fromIndex: Int, toIndex: Int, size: Int) {
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/collections/Arrays.kt]: Not sure if removing inline here is a good idea.

File: runtime/src/main/kotlin/kotlin/collections/Collections.kt [hunk 2]
```diff
@@ -11,11 +11,9 @@ import kotlin.random.*
 
 /** Copies typed varargs array to an array of objects */
 internal actual fun <T> Array<out T>.copyToArrayOfAny(isVarargs: Boolean): Array<out Any?> =
-        if (isVarargs)
-            // if the array came from varargs and already is array of Any, copying isn't required.
-            @Suppress("UNCHECKED_CAST") (this as Array<out Any?>)
-        else
-            @Suppress("UNCHECKED_CAST") (this.copyOfUninitializedElements(this.size) as Array<out Any?>)
+        // if the array came from varargs and already is array of Any, copying isn't required.
+        if (isVarargs) this
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/collections/Collections.kt]: Same line?

File: runtime/src/main/kotlin/kotlin/collections/Maps.kt [hunk 3]
```diff
@@ -6,7 +6,7 @@
 package kotlin.collections
 
 // creates a singleton copy of map, if there is specialization available in target platform, otherwise returns itself
-internal inline actual fun <K, V> Map<K, V>.toSingletonMapOrSelf(): Map<K, V> = toSingletonMap()
+internal actual fun <K, V> Map<K, V>.toSingletonMapOrSelf(): Map<K, V> = toSingletonMap()
```
Comment (olonho) [inline runtime/src/main/kotlin/kotlin/collections/Maps.kt]: Why remove inline here?

File: runtime/src/main/kotlin/kotlin/collections/Arrays.kt (hunk 1 above)
Comment (PavelPunegov) [reply runtime/src/main/kotlin/kotlin/collections/Arrays.kt]: Yes, it should not be removed. But I found that kotlin-compiler issues a warning here stating that inlining in this case makes performance impact insignificant.

File: runtime/src/main/kotlin/kotlin/collections/Maps.kt (hunk 3 above)
Comment (PavelPunegov) [reply runtime/src/main/kotlin/kotlin/collections/Maps.kt]: Here is the same as previous

File: runtime/src/main/kotlin/kotlin/collections/Collections.kt (hunk 2 above)
Comment (PavelPunegov) [reply runtime/src/main/kotlin/kotlin/collections/Collections.kt]: yep

Comment (olonho) [reply runtime/src/main/kotlin/kotlin/collections/Collections.kt]: I mean whole statement on same line.

**Pull request merged**